src/main/java/com/hotels/service/tracing/zipkintohaystack/forwarders/haystack/HaystackDomainConverter.java

```diff
@@ -48,12 +49,21 @@ public static Span fromZipkinV2(zipkin2.Span zipkin) {
         doIfNotNull(zipkin.parentId(), builder::setParentSpanId);
         doIfNotNull(zipkin.localServiceName(), builder::setServiceName);
 
-        if (!isEmpty(zipkin.tags())) {
-            zipkin.tags().forEach((key, value) -> {
-                List<Tag> tagStream = fromZipkinTag(key, value);
-                builder.addAllTags(tagStream);
-            });
-        }
+        List<Tag> tags = zipkin.tags().entrySet().stream()
+                .flatMap(entry -> fromZipkinTag(entry.getKey(), entry.getValue()).stream())
+                .collect(toList());
+        builder.addAllTags(tags);
+
+        var logs = zipkin.annotations().stream()
+                .map(annotation -> Log.newBuilder()
+                        .setTimestamp(annotation.timestamp())
+                        .addFields(Tag.newBuilder()
+                                .setKey("annotation")
```

Comment:
@worldtiki could you please link the line in the agent where this is happening?

Reply:
of course. i should have done that, sorry

https://github.com/ExpediaDotCom/haystack-agent/blob/master/agent-providers/span/src/main/java/com/expedia/www/haystack/agent/pitchfork/processors/HaystackDomainConverter.java#L65-L72